public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx

```diff
@@ -0,0 +1,340 @@
+      include_aliases: snapshot?.restore_aliases ? snapshot.restore_aliases : true,
+      partial: snapshot?.partial || false,
+    };
+    let repoError = "";
+
+    if (!snapshotId.trim()) {
+      this.setState({ snapshotIdError: "Required." });
+
+      return;
+    }
+    if (!selectedRepoValue) {
+      repoError = ERROR_PROMPT.REPO;
+      this.setState({ repoError });
+
+      return;
+    }
+    restoreSnapshot(snapshotId, selectedRepoValue, options);
+  };
+
+  onClickIndices = () => {
+    this.setState({ listIndices: true });
+  };
+
+  onIndicesSelectionChange = (selectedOptions: EuiComboBoxOptionOption<IndexItem>[]) => {
+    const selectedIndexOptions = selectedOptions.map((o) => o.label);
+    let newJSON = this.state.snapshot;
+    newJSON!.indices = [...selectedIndexOptions];
+    this.setState({ snapshot: newJSON, selectedIndexOptions: selectedOptions });
+  };
+
+  getSnapshot = async (snapshotId: string, repository: string) => {
+    const { snapshotManagementService } = this.props;
+
+    try {
+      const response = await snapshotManagementService.getSnapshot(snapshotId, repository);
+
+      if (response.ok) {
+        const newOptions = response.response.indices.map((index) => {
+          return { label: index };
+        });
+        this.setState({ snapshot: response.response, indexOptions: [...newOptions] });
+      }
+    } catch (err) {
+      this.context.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem loading the snapshot."));
+    }
+  };
+
+  getIndexOptions = () => {
+    this.getSnapshot(this.props.snapshotId, this.state.selectedRepoValue);
+  };
+
+  onCreateOption = (searchValue: string, options: Array<EuiComboBoxOptionOption<IndexItem>>) => {
+    const normalizedSearchValue = searchValue.trim().toLowerCase();
+    if (!normalizedSearchValue) {
+      return;
+    }
+    const newOption = {
+      label: searchValue,
+    };
+    // Create the option if it doesn't exist.
+    if (options.findIndex((option) => option.label.trim().toLowerCase() === normalizedSearchValue) === -1) {
+      this.setState({ indexOptions: [...this.state.indexOptions, newOption] });
+    }
+
+    const selectedIndexOptions = [...this.state.selectedIndexOptions, newOption];
+    this.setState({ selectedIndexOptions: selectedIndexOptions });
+  };
+
+  getRepos = async () => {
+    try {
+      const { snapshotManagementService } = this.props;
+      const response = await snapshotManagementService.catRepositories();
+      if (response.ok) {
+        const selectedRepoValue = response.response.length > 0 ? response.response[0].id : "";
+        this.setState({ repositories: response.response, selectedRepoValue });
+      } else {
+        this.context.notifications.toasts.addDanger(response.error);
+      }
+    } catch (err) {
+      this.context.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem loading the snapshots."));
+    }
+  };
```

Comment:
You can get repo value from Snapshots.tsx page and use that.

Reply:
Fixed, thanks Bowen!  removed getRepos, repository is now a prop